### DISCUSSION TITLE
Fix spacing in the sentence.

### DISCUSSION THREAD

**Discussion by dtemnov** [explosion/spaCy#10947] [Help: Coding & Implementations] [answered]
Hi!

I'm trying to code some simple preprocessing for my data and here is the use case:
"I know,you 're here!And me too" -> "I know, you're here! And me too."

Is it possible to do with spacy? I mean, to add additional spaces between the punctuation tokens?

**Comment by pmbaumgartner** [comment, marked as answer]
This isn't possible directly with spaCy, but you could use something like [symspell](https://pypi.org/project/symspellpy/) to help. For example, when I pass your string into the [basic usage](https://symspellpy.readthedocs.io/en/latest/examples/lookup_compound.html) example, it returns: `i know you re here and me too`, so it's close but with the punctuation removed. There may be a way to keep the punctuation and casing - check out this [open issue](https://github.com/mammothb/symspellpy/issues/80) for more details.

That might also be overkill if this only happens with punctuation. If that's the case, there is probably a regular expression you can create to add spaces when that punctuation appears in the middle of two words.
